### ISSUE TITLE
`pl` autocompletition on stable?

### ISSUE THREAD

**Issue by Kivooeo** [rust-lang/rust-analyzer#15802]
Why this suggest me nightly(?) feature on a stable version
![image](https://github.com/rust-lang/rust-analyzer/assets/75776246/feb87937-709a-43c3-92fe-3b0fdc6e0c93)

It should suggest me `println!` not a `place!`

`rust-analyzer --verstion` output
```
rust-analyzer --version
rust-analyzer 1.73.0 (cc66ad46 2023-10-03)
```
Previously, about two weeks ago, it suggests a `println!` on first place, i dont think this is correct work.

**Comment by lnicola** [comment]
Until two weeks ago we didn't suggest `place!` at all. But I agree, it shouldn't be the first suggestion.

**Comment by Kivooeo** [comment]
Any updates?

**Comment by Veykril** [comment]
https://github.com/rust-lang/rust-analyzer/blob/11a87c917943dac5a568579f799c2d7458324103/crates/ide-completion/src/completions/flyimport.rs#L267 and https://github.com/rust-lang/rust-analyzer/blob/11a87c917943dac5a568579f799c2d7458324103/crates/ide-completion/src/completions/flyimport.rs#L309
are the two places where we need to look at the stability of all segments of the `import.import_path`.

**Comment by lnicola** [comment]
Tentative, but non-working patch at https://github.com/lnicola/rust-analyzer/commit/873d34be1304b3d0e169bfe1c663a98670aa02d0.

**Comment by Veykril** [comment]
Actually, after having worked on https://github.com/rust-lang/rust-analyzer/pull/15871 just now, I think we can forego looking at the attributes entirely for this in completions again. We know the stability of an import path when calculating it in `find_path`, we just aren't exposing that from `find_path_inner` where we map the stability away. `LocatedImport`'s import paths being looked for here https://github.com/rust-lang/rust-analyzer/blob/ba61766217444d980a3a20f378181e2ad388ab0e/crates/ide-db/src/imports/import_assets.rs#L603-L613

**Comment by lnicola** [comment]
But my proposed change (which doesn't actually work for `place!` for some reason) also applies outside of flyimport. For example (this won't compile, of course),

```rust
struct S;

mod m {
    #![unstable]

    impl super::S {
        pub fn foo(&self) {}
    }
}
```

```
fn main() {
    S.<|>;
```

**Comment by Veykril** [comment]
Right, my idea won't cover method calls. Aside from that I don't think there really is a point in checking the parent module of anything though, since any other completion would require the thing in question to already be in scope, and hence the user should have the feature enabled anyways?

Walking up the ancestors also has the problem of re-exports through unstable modules not being recognized as unstable actually. (I think rustc itself currently fails this as well, or used to iirc)
